src/tag.rs

```diff
@@ -236,6 +262,44 @@ impl Op1 {
     pub fn as_field<F: LurkField>(&self) -> F {
         F::from(*self as u64)
     }
+
+    pub fn symbol_name(&self) -> &'static str {
+        match self {
+            Op1::Car => "CAR",
+            Op1::Cdr => "CDR",
+            Op1::Atom => "ATOM",
+            Op1::Emit => "EMIT",
+            Op1::Open => "OPEN",
+            Op1::Secret => "SECRET",
+            Op1::Commit => "COMMIT",
+            Op1::Num => "NUM",
+            Op1::Comm => "COMM",
+            Op1::Char => "CHAR",
+            Op1::Eval => "EVAL",
+            Op1::U64 => "U64",
+        }
+    }
+
+    pub fn all() -> Vec<Self> {
+        vec![
+            Op1::Car,
+            Op1::Cdr,
+            Op1::Atom,
+            Op1::Emit,
+            Op1::Open,
+            Op1::Secret,
+            Op1::Commit,
+            Op1::Num,
+            Op1::Comm,
+            Op1::Char,
+            // Op1::Eval,
```

Comment:
Why eval is commented out?

Reply:
It's because this ends up being used to test whether only a single argument should be accepted, and eval can take a second argument. Probably all of this can be refactored a bit in light of how things ended up. Good catch, thanks.